superset-frontend/src/welcome/DashboardTable.tsx

```diff
@@ -17,35 +17,45 @@
  * under the License.
  */
 import React from 'react';
-import PropTypes from 'prop-types';
 import { t } from '@superset-ui/translation';
 import { SupersetClient } from '@superset-ui/connection';
 import moment from 'moment';
 import { debounce } from 'lodash';
 import ListView from 'src/components/ListView/ListView';
 import withToasts from 'src/messageToasts/enhancers/withToasts';
+import { Dashboard } from 'src/types/bootstrapTypes';
+import { FetchDataConfig } from 'src/components/ListView/types';
 
 const PAGE_SIZE = 25;
 
-class DashboardTable extends React.PureComponent {
-  static propTypes = {
-    addDangerToast: PropTypes.func.isRequired,
-    search: PropTypes.string,
-  };
+interface DashboardTableProps {
```

Comment:
airbnb guide use `type` for `Props` instead of `interface`.

Reply:
oops... Well throughout Superset we've got 44 `interface` Props and 10 `type` Props. I'll go through and clean them all up (and try to find a lint rule) sometime in the future. For now I'd rather stick to the more common choice.

Reply:
I'm not sure about this recommendation. I like how interfaces gives you an alias and really makes intellisense and error messages more readable. Even TypeScript's official doc uses Interface: https://www.typescriptlang.org/docs/handbook/jsx.html#function-component

We have used interface in a lot of places and I have yet to see any problem.

Reply:
The point is not the code is not working. Both works.

Using `type` has convenience when hovering and you can see its content immediately because `type` is just an alias while `interface` encapsulates its content so you have to go to the definition to see its fields.

Oftentimes you want to see what can be in the `Props`.

when using type

![image](https://user-images.githubusercontent.com/1659771/87573999-94a34e80-c682-11ea-9bd8-821e18d4df19.png)

when using interface

![image](https://user-images.githubusercontent.com/1659771/87574025-9ff67a00-c682-11ea-97f7-4e97528275b5.png)

Reply:
Generally, when the call can go either way (both with pros/cons or none have good reasons), I would lean on consistency if it is not expensive change and no clear win. Otherwise the entire codebase will have many different dialects. That is probably worse.

If you think the benefit above is not convincing enough and the majority in this repo is already using `interface`, we can also start documenting these decisions for `incubator-superset`, and adopt `interface` for this repo.

For the internal Airbnb project, I'll stick to the TS style guide from the TS working group.